base/reducedim.jl

```diff
@@ -850,15 +976,44 @@ end
 isgreater(a, b) = isless(b,a)
 
 """
-    findmax!(rval, rind, A) -> (maxval, index)
+    findmax!(rval, rind, A; [reset::Bool = true]) -> (maxval, index)
 
 Find the maximum of `A` and the corresponding linear index along singleton
 dimensions of `rval` and `rind`, and store the results in `rval` and `rind`.
 `NaN` is treated as greater than all other values.
+If `reset` is `true`, values in `rval` and `rind` are discarded.  Otherwise, they are used
+as the initial value of the accumulation.
+
+!!! compat "Julia 1.6"
+    keyword argument `reset` is added in Julia 1.6.
+
+# Examples
+```jldoctest
+julia> A = [1.0 2; 3 4]
+2×2 Array{Float64,2}:
+ 1.0  2.0
+ 3.0  4.0
+
+julia> findmax!([5.0; 5.0], [CartesianIndex(-1, -1); CartesianIndex(-1, -1)], A)
+([2.0; 4.0], CartesianIndex{2}[CartesianIndex(1, 2); CartesianIndex(2, 2)])
+
+julia> findmin!([5.0  5.0], [CartesianIndex(-1, -1) CartesianIndex(-1, -1)], A)
+([3.0 4.0], CartesianIndex{2}[CartesianIndex(2, 1) CartesianIndex(2, 2)])
+
+julia> findmax!([5.0  5.0], [CartesianIndex(-1, -1) CartesianIndex(-1, -1)], A; reset = false)
+([5.0 5.0], CartesianIndex{2}[CartesianIndex(-1, -1) CartesianIndex(-1, -1)])
+```
 """
 function findmax!(rval::AbstractArray, rind::AbstractArray, A::AbstractArray;
-                  init::Bool=true)
-    findminmax!(isgreater, init && !isempty(A) ? fill!(rval, first(A)) : rval, fill!(rind,zero(eltype(keys(A)))), A)
+                  kw...)
+    if haskey(kw, :init)  # backward compatibility
+        _kw_reset(kw) && !isempty(A) && fill!(rval, first(A))
+        fill!(rind, zero(eltype(keys(A))))
+    elseif _kw_reset(kw)
+        !isempty(A) && fill!(rval, first(A))
+        fill!(rind, zero(eltype(keys(A))))
+    end
```

Comment:
The behavior `init=false` was not consistent with other multi-dimensional reducers.  So I suggest to fix it.

It was added in https://github.com/JuliaLang/julia/pull/6894 by @timholy but it looks like it was an internal function back then. So, probably the consistency with other reducers and the usability for end-users were not the biggest concern.

Also, I'm not sure if we should keep the backward compatibility branch. But it's rather cheap so why not?